In [1]:
import pandas as pd
import seaborn as sns
import matplotlib as plt
import numpy as np

# Import Data

First I import the data and print the shape and head to have a visual of what it looks like.

In [2]:
Audi = pd.read_csv("../desktop/ironhack/Cars/audi.csv", encoding="latin1")
print("Audi shape:",Audi.shape)
print(Audi.head(2))


Audi shape: (10668, 9)
  model  year  price transmission  mileage fuelType  tax   mpg  engineSize
0    A1  2017  12500       Manual    15735   Petrol  150  55.4         1.4
1    A6  2016  16500    Automatic    36203   Diesel   20  64.2         2.0


In [3]:
BMW = pd.read_csv("../desktop/ironhack/Cars/bmw.csv", encoding="latin1")
print("BMW shape:", BMW.shape)
print(BMW.head(2))


BMW shape: (10781, 9)
      model  year  price transmission  mileage fuelType  tax   mpg  engineSize
0  5 Series  2014  11200    Automatic    67068   Diesel  125  57.6         2.0
1  6 Series  2018  27000    Automatic    14827   Petrol  145  42.8         2.0


In [4]:
Ford = pd.read_csv("../desktop/ironhack/Cars/ford.csv", encoding="latin1")
print("Ford shape:",Ford.shape)
print(Ford.head(2))

Ford shape: (17965, 9)
     model  year  price transmission  mileage fuelType  tax   mpg  engineSize
0   Fiesta  2017  12000    Automatic    15944   Petrol  150  57.7         1.0
1    Focus  2018  14000       Manual     9083   Petrol  150  57.7         1.0


In [5]:
Toyota = pd.read_csv("../desktop/ironhack/Cars/toyota.csv", encoding="latin1")
print("Toyota shape:", Toyota.shape)
print(Toyota.head(2))

Toyota shape: (6738, 9)
  model  year  price transmission  mileage fuelType  tax   mpg  engineSize
0  GT86  2016  16000       Manual    24089   Petrol  265  36.2         2.0
1  GT86  2017  15995       Manual    18615   Petrol  145  36.2         2.0


In [6]:
Hyundai = pd.read_csv("../desktop/ironhack/Cars/hyundai.csv", encoding="latin1")
print("Hyundai shape:",Hyundai.shape)
print(Hyundai.head(2))

Hyundai shape: (4860, 9)
     model  year  price transmission  mileage fuelType  tax(£)   mpg  \
0      I20  2017   7999       Manual    17307   Petrol      145  58.9   
1   Tucson  2016  14499    Automatic    25233   Diesel      235  43.5   

   engineSize  
0         1.2  
1         2.0  


In [7]:
Mercedes = pd.read_csv("../desktop/ironhack/Cars/mercedes.csv", encoding="latin1")
print("Mercedes shape:", Mercedes.shape)
print(Mercedes.head(2))

Mercedes shape: (13119, 9)
      model  year  price transmission  mileage fuelType  tax   mpg  engineSize
0       SLK  2005   5200    Automatic    63000   Petrol  325  32.1         1.8
1   S Class  2017  34948    Automatic    27000   Hybrid   20  61.4         2.1


In [8]:
Skoda = pd.read_csv("../desktop/ironhack/Cars/skoda.csv", encoding="latin1")
print("Skoda shape", Skoda.shape)
print(Skoda.head(2))

Skoda shape (6267, 9)
      model  year  price transmission  mileage fuelType  tax   mpg  engineSize
0   Octavia  2017  10550       Manual    25250   Petrol  150  54.3         1.4
1    Citigo  2018   8200       Manual     1264   Petrol  145  67.3         1.0


In [9]:
Vauxhall = pd.read_csv("../desktop/ironhack/Cars/vauxhall.csv", encoding="latin1")
print("Vauxhall shape:",Vauxhall.shape)
print(Vauxhall.head(2))

Vauxhall shape: (13632, 9)
    model  year  price transmission  mileage fuelType  tax   mpg  engineSize
0   Corsa  2018   7885       Manual     9876   Petrol  145  55.4         1.4
1   Corsa  2019  11995       Manual     2500   Petrol  145  54.3         1.4


In [10]:
Volkswagen = pd.read_csv("../desktop/ironhack/Cars/Volkswagen.csv", encoding="latin1")
print("Volkswagen shape:", Volkswagen.shape)
print(Volkswagen.head(2))

Volkswagen shape: (15157, 9)
    model  year  price transmission  mileage fuelType  tax   mpg  engineSize
0   T-Roc  2019  25000    Automatic    13904   Diesel  145  49.6         2.0
1   T-Roc  2019  26883    Automatic     4562   Diesel  145  49.6         2.0


I have a data sets labeled "unclean_focus". I will look at it and see if has any use.

In [11]:
unclean_focus = pd.read_csv("../desktop/ironhack/Cars/unclean focus.csv", encoding="latin1")
print(unclean_focus.shape)
print(unclean_focus.head(2))

(5604, 11)
    model    year      price transmission mileage fuel type engine size  \
0   Focus  2016.0    £8,000       Manual     NaN    Petrol       0.999   
1   Focus  2019.0   £13,400       Manual     NaN    Petrol       0.999   

  mileage2 fuel type2 engine size2     reference  
0   38,852        NaN          NaN  /ad/25451236  
1   11,952        NaN          NaN  /ad/25466927  


# Cleaning 

The Ford Focus is a very popular car in the UK, so I will clean the data to use it.

In [12]:
unclean_focus.head()

model    year      price transmission mileage fuel type engine size  \
0   Focus  2016.0    £8,000       Manual     NaN    Petrol       0.999   
1   Focus  2019.0   £13,400       Manual     NaN    Petrol       0.999   
2   Focus  2019.0   £14,600       Manual     NaN    Petrol       1.498   
3   Focus  2016.0    £9,450       Manual     NaN    Diesel        1.56   
4   Focus  2015.0    £9,999       Manual     NaN    Diesel        1.56   

  mileage2 fuel type2 engine size2     reference  
0   38,852        NaN          NaN  /ad/25451236  
1   11,952        NaN          NaN  /ad/25466927  
2   22,142        NaN          NaN  /ad/25484363  
3   14,549        NaN          NaN  /ad/20444868  
4    7,010        NaN          NaN  /ad/22551057

In [13]:
unclean_focus['mileage'].unique()

array([nan, '13,255', '42,112', ..., '21,215', '1,478', '9,137'],
      dtype=object)

In [14]:
unclean_focus['mileage2'].unique()[:10]

array(['38,852', '11,952', '22,142', '14,549', '7,010', '26,796',
       '16,128', '4,900', '30,953', '3,000'], dtype=object)

In [15]:
unclean_focus = unclean_focus.drop(columns=['mileage','fuel type2', 'engine size2', 'reference'])

In [16]:
unclean_focus.head()

model    year      price transmission fuel type engine size mileage2
0   Focus  2016.0    £8,000       Manual    Petrol       0.999   38,852
1   Focus  2019.0   £13,400       Manual    Petrol       0.999   11,952
2   Focus  2019.0   £14,600       Manual    Petrol       1.498   22,142
3   Focus  2016.0    £9,450       Manual    Diesel        1.56   14,549
4   Focus  2015.0    £9,999       Manual    Diesel        1.56    7,010

In [17]:
print("model:",unclean_focus['model'].unique())
print("year:",unclean_focus['year'].unique())
print("price:",unclean_focus['price'].unique())
print("transmission:",unclean_focus['transmission'].unique())
print("fuel type",unclean_focus['fuel type'].unique())
print("engine size:",unclean_focus['engine size'].unique())
print("mileage:",unclean_focus['mileage2'].unique())

model: [' Focus' nan]
year: [2016. 2019. 2015. 2018. 2020. 2017. 2013. 2009. 2011. 2014.   nan 2012.
 2008. 2006. 2010. 2007. 2003. 2005. 2002.]
price: [' £8,000' ' £13,400' ' £14,600' ... ' £6,494' ' £8,200' ' £17,799']
transmission: ['Manual' 'Automatic' 'Semi-Auto' nan]
fuel type ['Petrol' 'Diesel' nan '11' '14' '26' '12' '24' '10' '13' '15' '33' '36'
 '35' '16' '22' '40' '8' '21' '9' '20' '7' '19' '34' '18' '25' '17' '31'
 '38' '6']
engine size: ['0.999' '1.498' '1.56' '999' '1000' '2300' '1498' '1.5' '£145' '£20'
 '£0' '£165' '£150' '£125' '2' '1' '£30' '£205' '2.3' '£235'
 '£160' '£145  ' '£150  ' '£200' '£240' nan '£260' '£120' '£140'
 '1596' '1499' '£300' '£195' '£325' '1999' '1.6' '1560' '£190' '£265'
 '£330' '£155' '£230' ' 2,00 cc']
mileage: ['38,852' '11,952' '22,142' '14,549' '7,010' '26,796' '16,128' '4,900'
 '30,953' '3,000' '57.7' '61.4' '74.3' '44.8' '80.7' '60.1' '67.3' '51.4'
 '8,879' '4,698' '16,199' '3,212' '8,201' '13,471' '12,372' '

In [18]:
for col in unclean_focus.columns:
    pct_missing = np.mean(unclean_focus[col].isnull())
    print('{} - {}%'.format(col, round(pct_missing*100)))

model - 1%
year - 3%
price - 1%
transmission - 1%
fuel type - 61%
engine size - 2%
mileage2 - 2%


There are a lot of missing values in fuel type, 61%, because it is an important factor in the price of a car, I will not drop the column, but I will drop the rows with the missing values.

In [19]:
unclean_focus.isnull().sum()

model             56
year             145
price             56
transmission      56
fuel type       3416
engine size      101
mileage2          95
dtype: int64

In [20]:
unclean_focus = unclean_focus.dropna()

In [21]:
unclean_focus.isnull().sum()

model           0
year            0
price           0
transmission    0
fuel type       0
engine size     0
mileage2        0
dtype: int64

In [22]:
unclean_focus['price'] = unclean_focus['price'].str.replace(r'\Â', '')
unclean_focus['price'] = unclean_focus['price'].str.replace(r'\£', '')
unclean_focus['engine size'] = unclean_focus['engine size'].str.replace(r'\Â', '')
unclean_focus['engine size'] = unclean_focus['engine size'].str.replace(r'\£', '')

<ipython-input-22-90a36d529bdb>:1: FutureWarning: The default value of regex will change from True to False in a future version.
  unclean_focus['price'] = unclean_focus['price'].str.replace(r'\Â', '')
<ipython-input-22-90a36d529bdb>:2: FutureWarning: The default value of regex will change from True to False in a future version.
  unclean_focus['price'] = unclean_focus['price'].str.replace(r'\£', '')
<ipython-input-22-90a36d529bdb>:3: FutureWarning: The default value of regex will change from True to False in a future version.
  unclean_focus['engine size'] = unclean_focus['engine size'].str.replace(r'\Â', '')
<ipython-input-22-90a36d529bdb>:4: FutureWarning: The default value of regex will change from True to False in a future version.
  unclean_focus['engine size'] = unclean_focus['engine size'].str.replace(r'\£', '')


In [23]:
clean_focus = unclean_focus
clean_focus

model    year    price transmission fuel type engine size mileage2
0      Focus  2016.0    8,000       Manual    Petrol       0.999   38,852
1      Focus  2019.0   13,400       Manual    Petrol       0.999   11,952
2      Focus  2019.0   14,600       Manual    Petrol       1.498   22,142
3      Focus  2016.0    9,450       Manual    Diesel        1.56   14,549
4      Focus  2015.0    9,999       Manual    Diesel        1.56    7,010
...      ...     ...      ...          ...       ...         ...      ...
5599   Focus  2019.0   18,745       Manual    Diesel           2    7,855
5600   Focus  2019.0   16,350       Manual    Petrol           1   13,891
5601   Focus  2019.0   16,850       Manual    Petrol           1   13,452
5602   Focus  2019.0   17,310    Automatic    Petrol           1   13,376
5603   Focus  2019.0   17,610    Automatic    Petrol           1    7,886

[2124 rows x 7 columns]

In [24]:
clean_focus = clean_focus.rename(columns={'mileage2': 'mileage', 'fuel type': 'fuelType', 'engine size':'engineSize'})

At the start the data set had 5604 rows and 11 columns, now that it is clean it has 2124 rows and 7 columns.

In [25]:
Audi.head()

model  year  price transmission  mileage fuelType  tax   mpg  engineSize
0    A1  2017  12500       Manual    15735   Petrol  150  55.4         1.4
1    A6  2016  16500    Automatic    36203   Diesel   20  64.2         2.0
2    A1  2016  11000       Manual    29946   Petrol   30  55.4         1.4
3    A4  2017  16800    Automatic    25952   Diesel  145  67.3         2.0
4    A3  2019  17300       Manual     1998   Petrol  145  49.6         1.0

In [26]:
to_drop = ['tax', 'mpg']

In [27]:
Audi.drop(to_drop, inplace=True, axis=1)

In [28]:
BMW.drop(to_drop, inplace=True, axis=1)

In [29]:
Ford.drop(to_drop, inplace=True, axis=1)

In [30]:
Toyota.drop(to_drop, inplace=True, axis=1)

In [31]:
Mercedes.drop(to_drop, inplace=True, axis=1)

In [32]:
Hyundai.drop(columns =['tax(£)', 'mpg'], inplace=True, axis=1)

In [33]:
Skoda.drop(to_drop, inplace=True, axis=1)

In [34]:
Vauxhall.drop(to_drop, inplace=True, axis=1)

In [35]:
Volkswagen.drop(to_drop, inplace=True, axis=1)

In [36]:
Audi['Brand'] = 'Audi'
BMW['Brand'] = 'BMW'
Ford['Brand'] = 'Ford'
Hyundai['Brand'] = 'Hyundai'
Mercedes['Brand'] = 'Mercedes'
Skoda['Brand'] = 'Skoda'
Toyota['Brand'] = 'Toyota'
Vauxhall['Brand'] = 'Vauxhall'
Volkswagen['Brand'] = 'Volkswagen'
clean_focus['Brand'] = 'Ford'

In [37]:
print(Audi.head(2))
print(BMW.head(2))
print(Ford.head(2))
print(Hyundai.head(2))
print(Mercedes.head(2))
print(Skoda.head(2))
print(Toyota.head(2))
print(Vauxhall.head(2))
print(Volkswagen.head(2))

  model  year  price transmission  mileage fuelType  engineSize Brand
0    A1  2017  12500       Manual    15735   Petrol         1.4  Audi
1    A6  2016  16500    Automatic    36203   Diesel         2.0  Audi
      model  year  price transmission  mileage fuelType  engineSize Brand
0  5 Series  2014  11200    Automatic    67068   Diesel         2.0   BMW
1  6 Series  2018  27000    Automatic    14827   Petrol         2.0   BMW
     model  year  price transmission  mileage fuelType  engineSize Brand
0   Fiesta  2017  12000    Automatic    15944   Petrol         1.0  Ford
1    Focus  2018  14000       Manual     9083   Petrol         1.0  Ford
     model  year  price transmission  mileage fuelType  engineSize    Brand
0      I20  2017   7999       Manual    17307   Petrol         1.2  Hyundai
1   Tucson  2016  14499    Automatic    25233   Diesel         2.0  Hyundai
      model  year  price transmission  mileage fuelType  engineSize     Brand
0       SLK  2005   5200    Automatic    63

In [38]:
clean_focus


model    year    price transmission fuelType engineSize mileage Brand
0      Focus  2016.0    8,000       Manual   Petrol      0.999  38,852  Ford
1      Focus  2019.0   13,400       Manual   Petrol      0.999  11,952  Ford
2      Focus  2019.0   14,600       Manual   Petrol      1.498  22,142  Ford
3      Focus  2016.0    9,450       Manual   Diesel       1.56  14,549  Ford
4      Focus  2015.0    9,999       Manual   Diesel       1.56   7,010  Ford
...      ...     ...      ...          ...      ...        ...     ...   ...
5599   Focus  2019.0   18,745       Manual   Diesel          2   7,855  Ford
5600   Focus  2019.0   16,350       Manual   Petrol          1  13,891  Ford
5601   Focus  2019.0   16,850       Manual   Petrol          1  13,452  Ford
5602   Focus  2019.0   17,310    Automatic   Petrol          1  13,376  Ford
5603   Focus  2019.0   17,610    Automatic   Petrol          1   7,886  Ford

[2124 rows x 8 columns]

In [39]:
frames = [Audi, BMW, Ford, Hyundai, Mercedes, Skoda, Toyota, Vauxhall, Volkswagen, clean_focus]

In [40]:
cars = pd.concat(frames)

In [41]:
cars = cars.rename(columns={'model': 'Model', 
                            'year': 'Year',
                            'price': 'Price',
                            'transmission': 'Transmission',
                            'mileage': 'Mileage',
                            'fuelType': 'FuelType',
                            'engineSize':'EngineSize'})

In [42]:
cars

Model    Year    Price Transmission Mileage FuelType EngineSize Brand
0         A1  2017.0    12500       Manual   15735   Petrol        1.4  Audi
1         A6  2016.0    16500    Automatic   36203   Diesel        2.0  Audi
2         A1  2016.0    11000       Manual   29946   Petrol        1.4  Audi
3         A4  2017.0    16800    Automatic   25952   Diesel        2.0  Audi
4         A3  2019.0    17300       Manual    1998   Petrol        1.0  Audi
...      ...     ...      ...          ...     ...      ...        ...   ...
5599   Focus  2019.0   18,745       Manual   7,855   Diesel          2  Ford
5600   Focus  2019.0   16,350       Manual  13,891   Petrol          1  Ford
5601   Focus  2019.0   16,850       Manual  13,452   Petrol          1  Ford
5602   Focus  2019.0   17,310    Automatic  13,376   Petrol          1  Ford
5603   Focus  2019.0   17,610    Automatic   7,886   Petrol          1  Ford

[101311 rows x 8 columns]

In [43]:
cars = cars[["Brand", "Model", "Year","EngineSize","Mileage", "Transmission", "FuelType", "Price"]]

In [44]:
cars

Brand   Model    Year EngineSize Mileage Transmission FuelType    Price
0     Audi      A1  2017.0        1.4   15735       Manual   Petrol    12500
1     Audi      A6  2016.0        2.0   36203    Automatic   Diesel    16500
2     Audi      A1  2016.0        1.4   29946       Manual   Petrol    11000
3     Audi      A4  2017.0        2.0   25952    Automatic   Diesel    16800
4     Audi      A3  2019.0        1.0    1998       Manual   Petrol    17300
...    ...     ...     ...        ...     ...          ...      ...      ...
5599  Ford   Focus  2019.0          2   7,855       Manual   Diesel   18,745
5600  Ford   Focus  2019.0          1  13,891       Manual   Petrol   16,350
5601  Ford   Focus  2019.0          1  13,452       Manual   Petrol   16,850
5602  Ford   Focus  2019.0          1  13,376    Automatic   Petrol   17,310
5603  Ford   Focus  2019.0          1   7,886    Automatic   Petrol   17,610

[101311 rows x 8 columns]

In [45]:
cars['FuelType'].unique()

array(['Petrol', 'Diesel', 'Hybrid', 'Other', 'Electric', '11', '14',
       '26', '12', '24', '10', '13', '15', '33', '36', '35', '16', '22',
       '40', '8', '21', '9', '20', '7', '19', '34', '18', '25', '17',
       '31', '38', '6'], dtype=object)

In [77]:
cars = cars.loc[cars['FuelType'] != '6']
cars

Brand   Model    Year EngineSize Mileage Transmission FuelType    Price
0     Audi      A1  2017.0        1.4   15735       Manual   Petrol    12500
1     Audi      A6  2016.0        2.0   36203    Automatic   Diesel    16500
2     Audi      A1  2016.0        1.4   29946       Manual   Petrol    11000
3     Audi      A4  2017.0        2.0   25952    Automatic   Diesel    16800
4     Audi      A3  2019.0        1.0    1998       Manual   Petrol    17300
...    ...     ...     ...        ...     ...          ...      ...      ...
5599  Ford   Focus  2019.0          2   7,855       Manual   Diesel   18,745
5600  Ford   Focus  2019.0          1  13,891       Manual   Petrol   16,350
5601  Ford   Focus  2019.0          1  13,452       Manual   Petrol   16,850
5602  Ford   Focus  2019.0          1  13,376    Automatic   Petrol   17,310
5603  Ford   Focus  2019.0          1   7,886    Automatic   Petrol   17,610

[100004 rows x 8 columns]

In [78]:
cars['FuelType'].unique()

array(['Petrol', 'Diesel', 'Hybrid', 'Other', 'Electric'], dtype=object)

In [79]:
cars['EngineSize'].unique()

array([1.4, 2.0, 1.0, 3.0, 1.6, 1.8, 1.5, 4.0, 2.5, 1.2, 4.2, 2.9, 5.2,
       1.9, 2.7, 0.0, 3.2, 4.1, 6.3, 0.6, 4.4, 2.2, 5.0, 6.6, 2.8, 3.5,
       1.1, 2.3, 1.3, 1.7, 2.4, 2.1, 5.5, 6.2, 4.7, 6.0, 3.7, 5.4, 4.3,
       4.5, '0.999', '1.498', '1.56', '999', '1000', '2300', '1498',
       '1.5', '2', '1', '2.3', '1596', '1499', '1.6', '1999', '1560',
       ' 2,00 cc'], dtype=object)

In [107]:
cars = cars.loc[cars['EngineSize'] != '0.0']
cars

Brand   Model    Year EngineSize  Mileage Transmission FuelType    Price
0     Audi      A1  2017.0   1.400000    15735       Manual   Petrol    12500
1     Audi      A6  2016.0   2.000000    36203    Automatic   Diesel    16500
2     Audi      A1  2016.0   1.400000    29946       Manual   Petrol    11000
3     Audi      A4  2017.0   2.000000    25952    Automatic   Diesel    16800
4     Audi      A3  2019.0   1.000000     1998       Manual   Petrol    17300
...    ...     ...     ...        ...      ...          ...      ...      ...
5355  Ford   Focus  2002.0        1.6  101,000       Manual   Petrol      795
5366  Ford   Focus  2019.0        1.5   12,086       Manual   Diesel   17,155
5463  Ford   Focus  2009.0    2,00 cc   77,368    Automatic   Diesel    2,995
5464  Ford   Focus  2009.0        1.6   97,498       Manual   Petrol    1,995
5582  Ford   Focus  2019.0          1   13,815    Automatic   Petrol   17,450

[97668 rows x 8 columns]

In [108]:
cars['EngineSize'].unique()

array([1.4, 2.0, 1.0, 3.0, 1.6, 1.8, 1.5, 4.0, 2.5, 1.2, 4.2, 2.9, 5.2,
       1.9, 2.7, 0.0, 3.2, 4.1, 6.3, 0.6, 4.4, 2.2, 5.0, 6.6, 2.8, 3.5,
       1.1, 2.3, 1.3, 1.7, 2.4, 2.1, 5.5, 6.2, 4.7, 6.0, 3.7, 5.4, 4.3,
       4.5, '0.999', '1.498', '1.56', '1.5', '2', '1', '2.3', '1.6',
       ' 2,00 cc'], dtype=object)

In [100]:
cars.duplicated().sum()

2312

In [104]:
cars = cars.drop_duplicates()

In [105]:
cars

Brand   Model    Year EngineSize  Mileage Transmission FuelType    Price
0     Audi      A1  2017.0   1.400000    15735       Manual   Petrol    12500
1     Audi      A6  2016.0   2.000000    36203    Automatic   Diesel    16500
2     Audi      A1  2016.0   1.400000    29946       Manual   Petrol    11000
3     Audi      A4  2017.0   2.000000    25952    Automatic   Diesel    16800
4     Audi      A3  2019.0   1.000000     1998       Manual   Petrol    17300
...    ...     ...     ...        ...      ...          ...      ...      ...
5355  Ford   Focus  2002.0        1.6  101,000       Manual   Petrol      795
5366  Ford   Focus  2019.0        1.5   12,086       Manual   Diesel   17,155
5463  Ford   Focus  2009.0    2,00 cc   77,368    Automatic   Diesel    2,995
5464  Ford   Focus  2009.0        1.6   97,498       Manual   Petrol    1,995
5582  Ford   Focus  2019.0          1   13,815    Automatic   Petrol   17,450

[97668 rows x 8 columns]

In [114]:
cars1 = cars.sample(frac=.5)
cars1

Brand     Model    Year EngineSize Mileage Transmission FuelType  \
5286       Toyota      Aygo  2017.0        1.0   10563    Automatic   Petrol   
10614        Audi        A4  2019.0        2.0    4080    Automatic   Petrol   
7500   Volkswagen      Polo  2017.0        1.0   35304       Manual   Petrol   
788        Toyota      RAV4  2017.0        2.5   26666    Automatic   Hybrid   
4555        Skoda      Yeti  2017.0        1.2   10345    Semi-Auto   Petrol   
...           ...       ...     ...        ...     ...          ...      ...   
13681  Volkswagen   Touareg  2016.0        3.0   18531    Semi-Auto   Diesel   
1354          BMW        M5  2019.0        4.4     123    Automatic   Petrol   
2527         Audi        A1  2015.0        1.4   26525       Manual   Petrol   
2532       Toyota     Yaris  2016.0        1.3   20788       Manual   Petrol   
2527        Skoda    Citigo  2018.0        1.0    9000       Manual   Petrol   

       Price  
5286    8000  
10614  26899  
7500    8999  
788    17500  
4555   14400  
...      ...  
13681  21990  
1354   68930  
2527   13495  
2532    9700  
2527    8199  

[48834 rows x 8 columns]

In [115]:
cars2 = cars.sample(frac=.3)
cars2

Brand         Model    Year EngineSize Mileage Transmission  \
8000     Vauxhall       Mokka X  2017.0        1.4   22832       Manual   
7466          BMW            X5  2014.0        3.0   97000    Automatic   
9025     Vauxhall       Mokka X  2018.0        1.4   11457       Manual   
10699    Mercedes       C Class  2013.0        2.1   57801    Automatic   
2259   Volkswagen          Golf  2015.0        1.4   28379       Manual   
...           ...           ...     ...        ...     ...          ...   
14866        Ford          Kuga  2016.0        2.0   53930    Automatic   
10231  Volkswagen        Tiguan  2019.0        2.0   12614    Automatic   
5959     Mercedes       C Class  2019.0        1.6      50    Automatic   
12427        Ford         Focus  2019.0        1.0    3788       Manual   
11918    Vauxhall   Grandland X  2019.0        1.5    2543    Automatic   

      FuelType  Price  
8000    Petrol  11498  
7466    Diesel  18990  
9025    Petrol  10954  
10699   Diesel   9700  
2259    Petrol  10898  
...        ...    ...  
14866   Diesel  13487  
10231   Diesel  24999  
5959    Diesel  37321  
12427   Petrol  16990  
11918   Diesel  28991  

[29300 rows x 8 columns]

In [106]:
cars.to_csv('data/cars.csv', header=True, index=False)

In [113]:
cars1.to_csv('data/cars1.csv', header=True, index=False)

In [116]:
cars2.to_csv('data/cars2.csv', header=True, index=False)